### PR TITLE
Attempted Week 1 Challenge with bugs fixed

### PR DESCRIPTION
**Pull request by goodnessemmanuel**
I did not test my previous code before creating PR in previous commit. I have fixed the bugs and updated my code.

### CODE REVIEW

**Comment by goodnessemmanuel** [comment]
@bondz 
is there something I'm not doing write here? 
Bond0
Bond2
Bond1
 have tried this:  
  **try  {
    if(typeof text !== "string") 
      throw `Please provide a morse string`;

    if(text === undefined) 
      throw `Please provide a morse string`;      
  }
  catch(err)  {
    return err;
  }**

and this:

**if (typeof of text === "string" || text === undefined || text === null ){
    return `Please provide a morse string`; 
}**
Still no luck . My browser (chrome) console logs the message as expected when I call '**console.log(morse()) or console.log(isolateDuplicates())**' but can't seem to figure it out on the GitHub test check(picture 1). Kindly help with the fix. Thank you

File: src/isolate-duplicates/index.js [hunk 1]
```diff
@@ -1,3 +1,76 @@
-function isolateDuplicates(text) {}
+function isolateDuplicates(text) 
+{
+    //if the parameter type is not a valid string throw an error 
+    try
+    {
+      if(typeof text !== "string") 
+        throw `Please enter a valid string`;
```
Comment (bondz) [inline src/isolate-duplicates/index.js:7]: You don't need to wrap this in a try...catch if you're expecting an error. Also, use braces.

File: src/morse/index.js [hunk 2]
```diff
@@ -58,6 +58,53 @@ const MORSE_CODE = {
...
+      throw `Please provide a morse string`;
 
-module.exports = morse;
+    if(text === undefined) 
+      throw `Please provide a morse string`;      
+  }
+  catch(err)
+  {
+    return err;
+  }
```
Comment (bondz) [inline src/morse/index.js:75]: Wrapping it `try..catch` means the error is thrown, catch and the function returns an error instead of throwing one.

Comment (goodnessemmanuel) [reply src/morse/index.js]: okay sir

File: src/isolate-duplicates/index.js [hunk 3]
```diff
@@ -1,3 +1,67 @@
-function isolateDuplicates(text) {}
+function isolateDuplicates(text) 
+{
+    //if the parameter type is not a valid string throw an error 
+    if(typeof text !== "string") 
+    {
+        throw `Please enter a valid string`
```
Comment (bondz) [inline src/isolate-duplicates/index.js:6]: Prefer to `throw Error(...)` rather than strings. Helps preserve the stack trace for debugging.

File: src/isolate-duplicates/index.js [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+    //loop through the input text to check valid input
+    for (let letter of text)
+    {
+        /*if you test any letter of the text and its not an alphabeth, throw 
+        *an error prompting the user to enter a valid input to the function */
+        if (!regexToValidateText.test(letter))
+        {
+            return `Please enter a valid string`;
+        }
+    }
```
Comment (bondz) [inline src/isolate-duplicates/index.js:20]: This doesn't actually throw an error. And there are prolly ways to validate the whole string is valid with regex without checking every character individually.

File: src/morse/index.js [hunk 5]
```diff
@@ -58,6 +58,45 @@ const MORSE_CODE = {
 
 Object.freeze(MORSE_CODE);
 
-function morse(text) {}
+function morse(text) 
+{
+  //check to ensure the input text is a valid else throw an error
+   if(typeof text !== "string")
+    {        
+      throw `Please provide a morse string`;
```
Comment (bondz) [inline src/morse/index.js:66]: Same as above.

File: src/morse/index.js [hunk 6]
```diff
@@ -58,6 +58,45 @@ const MORSE_CODE = {
...
+      {
+        decodeMorse.push(" ");//preserve all whitespace character in the input text
+      } 
+      else if(MORSE_CODE.hasOwnProperty(item))
+      {
+        decodeMorse.push(MORSE_CODE[item]);
+      }
+      else
+      {
+        return `Please provide a morse string`;
```
Comment (bondz) [inline src/morse/index.js:88]: You should rather throw than return an incorrect value.

File: src/remove-dulplicates/index.js [hunk 7]
```diff
@@ -1,3 +1,52 @@
...
+    let keys = Object.keys(obj);
+
+    //loop through the object keys referencing the values and perform the underly operation
+    for (let k = 0; k < keys.length; k++)
+    {
+        //within the value of each key that is an array, loop through the elements 
+        for (let i = 0; i < obj[keys[k]].length; i++)
+        {
+            //if any element has more than one index... 
+            if (obj[keys[k]].indexOf(obj[keys[k]][i]) !== i)
```
Comment (bondz) [inline src/remove-dulplicates/index.js:15]: All the nested braces make it hard to follow what's happening. You might want to extract to variables.
